src/Models/Services/LocalStorageManager/LocalStorageManager.ts

```diff
@@ -0,0 +1,606 @@
+                        AzureResourceTypes.DigitalTwinInstance
+                    )
+                );
+            }
+            // END of migration
+        }
+        return environmentOptionsInLocalStorage?.adtInstances;
+    } catch (error) {
+        console.error(error.message);
+        return null;
+    }
+};
+
+/** To get the list of Storage accounts from local storage to be used as options in EnvironmentPicker
+ * @return list of Storage accounts as environment item from local storage
+ */
+export const getStorageAccountOptionsFromLocalStorage = (): Array<EnvironmentItemInLocalStorage> | null => {
+    try {
+        const environmentOptionsInLocalStorage =
+            getEnvironmentOptionsFromLocalStorage() || {};
+        if (!environmentOptionsInLocalStorage?.storageAccounts) {
+            // START of migration of values using old local storage key
+            const previouslyUsedKey = StorageAccountsLocalStorageKey;
+
+            const oldOptionsInLocalStorage = localStorage.getItem(
+                previouslyUsedKey
+            );
+            if (oldOptionsInLocalStorage) {
+                const options = oldOptionsInLocalStorage
+                    ? (JSON.parse(
+                          oldOptionsInLocalStorage
+                      ) as Array<StorageAccountsInLocalStorage>).map(
+                          (e) =>
+                              ({
+                                  id: e.id,
+                                  name: getNameOfResource(
+                                      e.url,
+                                      AzureResourceTypes.StorageAccount
+                                  ),
+                                  properties: {
+                                      primaryEndpoints: { blob: e.url }
+                                  },
+                                  type: AzureResourceTypes.StorageAccount
+                              } as IAzureStorageAccount)
+                      )
+                    : null;
+
+                setStorageAccountOptionsInLocalStorage(options);
+                localStorage.removeItem(previouslyUsedKey);
+                return options.map((o) =>
+                    getEnvironmentItemFromResource(
+                        o,
+                        AzureResourceTypes.StorageAccount
+                    )
+                );
+            }
+            // END of migration
+        }
+        return environmentOptionsInLocalStorage?.storageAccounts;
+    } catch (error) {
+        console.error(error.message);
+        return null;
+    }
+};
+
+/** To update the list of Storage containers in local storage to be used as options in EnvironmentPicker
+ * @return list of Storage containers as environment item from local storage
+ */
+export const getStorageContainerOptionsFromLocalStorage = (
+    localStorageKey?: string
+): Array<EnvironmentItemInLocalStorage> | null => {
+    try {
+        const environmentOptionsInLocalStorage =
```

Comment:
In fact this line we are talking about in other comment does that: https://github.com/microsoft/iot-cardboard-js/pull/706/files/49fe990c2c2d73a6e992b2a4d2c9f6c38f4c297a#r986077499

Reply:
would it be possible to handle the migrations all in one spot? What migration are we doing here if not the one that brings in the state from ADT Explorer?

Reply:
the migration is being done in individual getter methods in all LocalStorageManager which I think make sense like whenever we attempt to get the new key-if not exist there do the migration first then return it. Migration is different that updating the old key for ADT explorer.

Reply:
hmmm ok. They just seem to be very similar since both are local storage. Could we not update the old key in ADT Explorer here so it all lives together and isn't likely to get out of sync?